### PR TITLE
Update Lisk SDK to v6.0.0-beta.4

### LINKED ISSUE #791: Update Lisk SDK to v6.0.0-beta.4
### Description

Update Lisk SDK to v6.0.0-beta.4.

### Acceptance Criteria

- Update SDK to v6.0.0-beta.4
- Update NodeJS to v18
  - package.json
  - .nvmrc
  - Dockerfile
  - snapcraft.yaml
- Update node configs:
  - Set `config.genesis.minimumCertifyHeight` (In accordance to the LIP updates on [GitHub PR](https://github.com/LiskHQ/lips/pull/381))
  - Set `supportedTokensSubstore` for the `token` module as per the changes specified in https://github.com/LiskHQ/lips/pull/378
  - Fix `token` module genesis assets for betanet
- Ensure Lisk Core builds and is able to start
- Tag Lisk Core v4.0.0-beta.2

### PR DESCRIPTION
**Pull request by sameersubudhi**
### What was the problem?

This PR resolves #791 #793 

### How was it solved?

- [x] Upgrade SDK to v6.0.0-beta.4
- [x] Upgrade NodeJS to v18
  - [x] .npmrc
  - [x] package.json
  - [x] Dockerfile (**NOTE**: Unable to build the images because of `@chainsafe/blst@0.2.6` dependency in SDK, should be fixed after SDK updates to `0.2.9`)
  - [x] snapcraft.yaml
- [x] Update node configs
  - [x] Set `pos.maxBFTWeightCap: 1000`
  - [x] Set config.genesis.minimumCertifyHeight (In accordance to the LIP updates on https://github.com/LiskHQ/lips/pull/381)
  - [x] Set supportedTokensSubstore for the token module as per the changes specified in https://github.com/LiskHQ/lips/pull/378
  - [x] Fix token module genesis assets for betanet
- [x] Configure chain-connector-plugin to run in-memory

### How was it tested?

Local

### CODE REVIEW

**Review by Nazgolze — APPROVED** [review]
lgtm

**Review by ManuGowda — COMMENTED** [review]
README needs to be updated with node.js version update?

File: README.md [hunk 1]
```diff
@@ -132,7 +132,13 @@ For a more advanced options refer to [PM2 documentation](https://pm2.keymetrics.
 
 [`lisk-core start`](docs/commands/start.md) supports flag and environment variable options to configure a node.
 
-Also, custom configuration through JSON file is available through the `--config, -c` flag.
+Also, custom configuration through JSON file can be made available through the `--config, -c` flag.
```
Comment (chris529) [inline README.md:135]:
```suggestion
In addition, custom configuration through the JSON file can be made available through the `--config, -c` flag.
```

**Review by ManuGowda — APPROVED** [review]

**Review by chris529 — APPROVED** [review]
LGTM 👍

**Pull request merged**